### PR TITLE
[SPARK-11338][WebUI] Prepend app links on HistoryPage with uiRoot path

### PR DESCRIPTION
**Pull request by ckadner**
[SPARK-11338: HistoryPage not multi-tenancy enabled ...](https://issues.apache.org/jira/browse/SPARK-11338)
- `HistoryPage.scala` ...prepending all page links with the web proxy (`uiRoot`) path
- `HistoryServerSuite.scala` ...adding a test case to verify all site-relative links are prefixed when the environment variable `APPLICATION_WEB_PROXY_BASE` (or System property `spark.ui.proxyBase`) is set

### CODE REVIEW

**Comment by BryanCutler** [comment]
I took a look, LGTM

**Comment by holdenk** [comment]
I've taken a first look, and it looks good pending tests. Could maybe @andrewor14 or @vanzin or @tsudukim take a look (since it seems like you are some of the more recent contributors to this part of the history server) / whitelist or trigger jenkins tests

**Comment by holdenk** [comment]
@ckadner one question is would it make sense to simply always prepend inside of makepagelink instead of doing this change everywhere? Looking at the other pages though it seems that most them do it this way so maybe there is a good reason.

**Comment by vanzin** [comment]
+1 to @holdenk's comment. If you're changing all invocations of `makePageLink` it might be easier to just change `makePageLink` itself.

**Comment by andrewor14** [comment]
ok to test

**Comment by ckadner** [comment]
@holdenk @vanzin, thank you for your comments! I just pushed an update to my fix.

**Comment by ckadner** [comment]
@vanzin, @andrewor14, if you are okay with this fix, do you think it could still make it into 1.6?

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala [hunk 1]
```diff
@@ -218,9 +217,9 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
   }
 
   private def makePageLink(linkPage: Int, showIncomplete: Boolean): String = {
-    "/?" + Array(
+    UIUtils.prependBaseUri("/?" + Array(
       "page=" + linkPage,
       "showIncomplete=" + showIncomplete
-    ).mkString("&")
+    ).mkString("&"))
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala]: nit: this line should be indented now.

File: core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ import org.scalatest.{BeforeAndAfter, Matchers}
 import org.scalatest.mock.MockitoSugar
 
 import org.apache.spark.{JsonTestUtils, SecurityManager, SparkConf, SparkFunSuite}
-import org.apache.spark.ui.SparkUI
+import org.apache.spark.ui.{UIUtils, SparkUI}
```
Comment (vanzin) [inline core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala]: nit: ordering

**Comment by vanzin** [comment]
Couple of nits, otherwise LGTM.

**Comment by ckadner** [comment]
@vanzin -- I pushed the nit pickity fixes :mag:  -- Thanks :-)

**Comment by vanzin** [comment]
Merging to master.

**Comment by ckadner** [comment]
@vanzin -- Thank you!

**Pull request closed without merging**